### PR TITLE
When TerraformBoot enables authentication, Xpanse calls its APIs alwa…

### LINKED ISSUE #1043: When TerraformBoot enables authentication, Xpanse calls its APIs always return  401 unauthenticated exception.
When TerraformBoot enables authentication, Xpanse calls its APIs always return  401 unauthenticated exception.
image
image

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #1043
related to https://github.com/eclipse-xpanse/terraform-boot/pull/37

image
image

image
image

### CODE REVIEW

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoCacheHolder.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * User information cache.
+ */
+@Profile("zitadel")
+public class CurrentUserInfoCacheHolder {
+
+    public static final String CURRENT_USERINFO_CACHE_NAME = "CurrentUserinfo";
+
+    private static final Map<String, CurrentUserInfo> USERINFO_CACHE
```
Comment (baixinsui) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoCacheHolder.java:21]: Using this solution, the map seems always store only the latest putted user . Please test if the user information obtained in the sub thread is correct if multiple users access it simultaneously. If not correct, you can try to use the `ThreadLocal<CurrentUserInfo>` to obtain the user information in sub threads.

File: modules/security/src/main/java/org/eclipse/xpanse/modules/security/zitadel/ZitadelIdentityProviderService.java [hunk 2]
```diff
@@ -172,6 +176,11 @@ public CurrentUserInfo getCurrentUserInfo() {
                     }
                 }
             }
+            OAuth2Token token =
+                    ((AbstractOAuth2TokenAuthenticationToken<?>) authentication).getToken();
+            currentUserInfo.setToken(
+                    ZitadelOauth2Constant.USER_TOKEN_PREFIX + token.getTokenValue());
+            CurrentUserInfoCacheHolder.getSingleton().setCurrentUserInfo(currentUserInfo);
```
Comment (baixinsui) [inline modules/security/src/main/java/org/eclipse/xpanse/modules/security/zitadel/ZitadelIdentityProviderService.java:183]: If current solution using `ConcurrentHashMap` works well,  please move the set action to `IdentityProviderManager`. If change to use the new solution `ConcurrentHashMap`, delete the set acition.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoCacheHolder.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            = new ConcurrentHashMap(0);
+
```
Comment (baixinsui) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoCacheHolder.java:23]: The initial capacity of map is `0`?

**Review by baixinsui — CHANGES_REQUESTED** [review]
Some comment need to check

**Review by baixinsui — APPROVED** [review]
LGTM

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoHolder.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+package org.eclipse.xpanse.modules.models.security.model;
+
+import java.util.Objects;
+import org.springframework.context.annotation.Profile;
+
+/**
+ * User information holder.
+ */
+@Profile("zitadel")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/security/model/CurrentUserInfoHolder.java:15]: do we need this profile annotation as this class is not a spring bean?

please also consider that if Zitdel profile is not enabled .. then there is no token available. So in this case, no auth header must be sent.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**